### PR TITLE
mon/OSDMonitor: tolerate upgrade from post-kraken dev cluster

### PR DESCRIPTION
**Pull request by liewegas**
If the 'creating' pgs key is missing, move on without crashing.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
@tchaikov this affects my test cluster

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -244,10 +244,16 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
 	ceph::features::mon::FEATURE_LUMINOUS)) {
     bufferlist bl;
     mon->store->get(OSD_PG_CREATING_PREFIX, "creating", bl);
-    auto p = bl.begin();
-    std::lock_guard<std::mutex> l(creating_pgs_lock);
-    creating_pgs.decode(p);
-    dout(7) << __func__ << " loading creating_pgs e" << creating_pgs.last_scan_epoch << dendl;
+    if (bl.length()) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: or we can compare the return code of `get()` with `-ENOENT`.

**Comment by yuriw** [comment]
test this please

**Comment by yuriw** [comment]
This was tested and can be merged

http://pulpito.ceph.com/yuriw-2017-04-13_22:21:45-rados-wip-yuri-testing_2017_4_15-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-04-13_22:21:45-rados-wip-yuri-testing_2017_4_15-distro-basic-smithi/

**Comment by jdurgin** [comment]
github's web ui conflict resolver made a merge commit - should be resolved manually

**Comment by liewegas** [comment]
rebased

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-04-21_23:56:14-rados-wip-sage-testing---basic-smithi/

**Comment by smithfarm** [comment]
Does this fix http://tracker.ceph.com/issues/19500 ?

**Comment by liewegas** [comment]
yeah

**Pull request merged**